### PR TITLE
Consolidate versions for easier extension

### PR DESCRIPTION
**Pull request by yuhongsun96**

### CODE REVIEW

File: backend/danswer/configs/app_configs.py [hunk 1]
```diff
@@ -62,18 +58,21 @@
 )
 
 # OAuth Login Flow
-ENABLE_OAUTH = os.environ.get("ENABLE_OAUTH", "").lower() != "false"
-OAUTH_TYPE = os.environ.get("OAUTH_TYPE", "google").lower()
 OAUTH_CLIENT_ID = os.environ.get(
-    "OAUTH_CLIENT_ID", os.environ.get("GOOGLE_OAUTH_CLIENT_ID", "")
+    "OAUTH_CLIENT_ID", os.environ.get("OAUTH_CLIENT_ID", "")
```
Comment (Weves) [inline backend/danswer/configs/app_configs.py]: breaking change :( can we keep things as is?

File: backend/danswer/configs/app_configs.py [hunk 2]
```diff
@@ -14,32 +15,27 @@
...
+WEB_DOMAIN = os.environ.get("WEB_DOMAIN") or "http://localhost:3000"
 
 
 #####
 # Auth Configs
 #####
-DISABLE_AUTH = os.environ.get("DISABLE_AUTH", "").lower() == "true"
-REQUIRE_EMAIL_VERIFICATION = (
-    os.environ.get("REQUIRE_EMAIL_VERIFICATION", "").lower() == "true"
+AUTH_TYPE = AuthType((os.environ.get("AUTH_TYPE") or AuthType.DISABLED.value).lower())
```
Comment (Weves) [inline backend/danswer/configs/app_configs.py:32]: Need to make sure we make an announcements about this

File: backend/danswer/configs/app_configs.py (hunk 1 above)
Comment (Weves) [reply backend/danswer/configs/app_configs.py]: Or at the very least tag everyone and tell them that this is changed

**Review by Weves — APPROVED** [review]

File: backend/danswer/configs/app_configs.py (hunk 1 above)
Comment (yuhongsun96) [reply backend/danswer/configs/app_configs.py]: actually I'm in favor of changing this back to Google OAuth :P We use a google specific library anyhow for this.

**Pull request merged**